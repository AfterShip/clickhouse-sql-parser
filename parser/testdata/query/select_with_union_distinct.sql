SELECT replica_name FROM system.ha_replicas UNION DISTINCT SELECT replica_name FROM system.ha_unique_replicas format JSON