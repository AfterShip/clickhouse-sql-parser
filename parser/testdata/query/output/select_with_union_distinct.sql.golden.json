[
  {
    "SelectPos": 0,
    "StatementEnd": 43,
    "With": null,
    "Top": null,
    "HasDistinct": false,
    "SelectItems": [
      {
        "Expr": {
          "Name": "replica_name",
          "QuoteType": 1,
          "NamePos": 7,
          "NameEnd": 19
        },
        "Modifiers": [],
        "Alias": null
      }
    ],
    "From": {
      "FromPos": 20,
      "Expr": {
        "Table": {
          "TablePos": 25,
          "TableEnd": 43,
          "Alias": null,
          "Expr": {
            "Database": {
              "Name": "system",
              "QuoteType": 1,
              "NamePos": 25,
              "NameEnd": 31
            },
            "Table": {
              "Name": "ha_replicas",
              "QuoteType": 1,
              "NamePos": 32,
              "NameEnd": 43
            }
          },
          "HasFinal": false
        },
        "StatementEnd": 43,
        "SampleRatio": null,
        "HasFinal": false
      }
    },
    "ArrayJoin": null,
    "Window": null,
    "Prewhere": null,
    "Where": null,
    "GroupBy": null,
    "WithTotal": false,
    "Having": null,
    "OrderBy": null,
    "LimitBy": null,
    "Limit": null,
    "Settings": null,
    "Format": null,
    "UnionAll": null,
    "UnionDistinct": {
      "SelectPos": 59,
      "StatementEnd": 121,
      "With": null,
      "Top": null,
      "HasDistinct": false,
      "SelectItems": [
        {
          "Expr": {
            "Name": "replica_name",
            "QuoteType": 1,
            "NamePos": 66,
            "NameEnd": 78
          },
          "Modifiers": [],
          "Alias": null
        }
      ],
      "From": {
        "FromPos": 79,
        "Expr": {
          "Table": {
            "TablePos": 84,
            "TableEnd": 109,
            "Alias": null,
            "Expr": {
              "Database": {
                "Name": "system",
                "QuoteType": 1,
                "NamePos": 84,
                "NameEnd": 90
              },
              "Table": {
                "Name": "ha_unique_replicas",
                "QuoteType": 1,
                "NamePos": 91,
                "NameEnd": 109
              }
            },
            "HasFinal": false
          },
          "StatementEnd": 109,
          "SampleRatio": null,
          "HasFinal": false
        }
      },
      "ArrayJoin": null,
      "Window": null,
      "Prewhere": null,
      "Where": null,
      "GroupBy": null,
      "WithTotal": false,
      "Having": null,
      "OrderBy": null,
      "LimitBy": null,
      "Limit": null,
      "Settings": null,
      "Format": {
        "FormatPos": 110,
        "Format": {
          "Name": "JSON",
          "QuoteType": 1,
          "NamePos": 117,
          "NameEnd": 121
        }
      },
      "UnionAll": null,
      "UnionDistinct": null,
      "Except": null
    },
    "Except": null
  }
]